### PR TITLE
feat(logs): deployment

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

[> Link to the JIRA ticket
](https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-547)

- First version of deployment logs

Capture d’écran 2022-11-16 à 11 19 06

---

### Global

- [x] This PR does not introduce any breaking change
- [ ] This PR introduces breaking change(s) and has been labeled as such
- [x] I have found someone to review this PR and pinged him

### Store

- [ ] This PR introduces new store changes

### NX

- [x] I have run the dep-graph locally and made sure the tree was clean i.e no circular dependencies
- [ ] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`

### Clean Code

- [x] I made sure the code is type safe (no any)
- [ ] I have included a feature flag on my feature, if applicable

### CODE REVIEW

**Comment by evoxmusic** [comment]
A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [[PR] staging - feat(logs): deployment - 2022-11-09T17:20:02Z](https://console.qovery.com/platform/organization/141c07c8-0dd9-4623-983b-3fdd61867255/projects/977bd069-20f7-4b24-bd95-28b44174cf43/environments/b9bf4664-f5a5-4190-9375-783488646d9e/applications)

 Another comment will be posted when deployments are terminated

File: libs/domains/environment/src/lib/mocks/deployment-log-factory.mock.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { Chance } from 'chance'
+import { EnvironmentLogs } from 'qovery-typescript-axios'
+import { LogsType } from '@qovery/shared/enums'
+
+const chance = new Chance()
+
+export const deploymentLogFactoryMock = (howMany: number, withError?: boolean): EnvironmentLogs[] =>
+  Array.from({ length: howMany }).map(() => ({
```
Comment (bdebon) [inline libs/domains/environment/src/lib/mocks/deployment-log-factory.mock.ts:8]: Old guys like me would have done a for(const i, i++) loop... That method is nice! I like it!

File: libs/pages/application/src/lib/ui/container/container.tsx [hunk 2]
```diff
@@ -97,13 +97,7 @@ export function Container(props: ContainerProps) {
                 {
                   name: 'Deployment logs',
                   contentLeft: <Icon name="icon-solid-scroll" className="text-brand-500 text-sm" />,
-                  onClick: () =>
```
Comment (bdebon) [inline libs/pages/application/src/lib/ui/container/container.tsx:100]: Farewell v2... <3

File: libs/pages/logs/deployment/src/lib/ui/row/row.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+  }
+
+  it('should render successfully', () => {
+    const { baseElement } = render(<Row {...props} />)
+    expect(baseElement).toBeTruthy()
+  })
+
+  it('should have success index color', () => {
+    props.data = {
+      type: 'info',
```
Comment (bdebon) [inline libs/pages/logs/deployment/src/lib/ui/row/row.spec.tsx:24]: Below you use LogsType.ERROR, can't you use LogsType.Info here?

File: libs/pages/logs/deployment/src/lib/page-deployment-logs.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    return new Promise((resolve) => {
+      environment?.cluster_id && resolve(url + `&bearer_token=${token}`)
+    })
+  }, [organizationId, environment?.cluster_id, projectId, environmentId, getAccessTokenSilently])
+
+  useWebSocket(deploymentLogsUrl, {
+    onMessage: (message) => {
+      setLoading('loaded')
+
+      if (pauseStatusLogs) {
```
Comment (bdebon) [inline libs/pages/logs/deployment/src/lib/page-deployment-logs.tsx:49]: eheheh yes !

File: libs/shared/ui/src/lib/components/tooltip/tooltip.tsx [hunk 5]
```diff
@@ -33,13 +33,18 @@ export function Tooltip(props: TooltipProps) {
     >
       <TooltipPrimitive.Trigger asChild>{children}</TooltipPrimitive.Trigger>
       <TooltipPrimitive.Content
-        className="bg-element-dark-400 text-text-100 rounded-sm px-2 py-1 text-xs font-medium"
+        className="bg-element-dark-400 text-text-100 dark:bg-element-light-lighter-200 dark:text-text-700 rounded-sm px-2 py-1 text-xs font-medium"
```
Comment (bdebon) [inline libs/shared/ui/src/lib/components/tooltip/tooltip.tsx:36]: So happy to see that...

**Review by bdebon — COMMENTED** [review]
So cool to be able to switch between deployment logs and logs.
A little bit sad that the Loading  screen remove the selected tab. It would be so nice to keep them and just refresh the logs. Hard to do?

CleanShot 2022-11-16 at 15 43 30@2x

Two things about these icons. There are not super readable and cute at this size, the spacing with the label is weird and a SVG duplicated for every log line can increase hugely the performances needed to run the logs. Why don't we remove them simply? What do you think?
CleanShot 2022-11-16 at 15 51 42@2x

**Pull request merged**